exporter/datadogexporter/examples/collector-metrics.yaml

```diff
@@ -17,7 +17,7 @@ exporters:
   datadog:
     api:
       site: ${env:DD_SITE}
-      key: ${env:DD_API_KEY}
+      key: ${DD_API_KEY}
```

Comment:
@mx-psi changing this from `${env:DD_API_KEY}` to `${DD_API_KEY}` breaks the test https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/12697854235/job/35395198770?pr=37122

```
exporters::datadog: api::key contains invalid characters: invalid characters: $, {, _, P, I, _, K, Y, }
```

this error indicates `${DD_API_KEY}` is not expanded